pkgs/applications/science/biology/bowtie2/default.nix

```diff
@@ -12,7 +12,7 @@ stdenv.mkDerivation rec {
     sha256 = "1zl3cf327y2p7p03cavymbh7b00djc7lncfaqih33n96iy9q8ibp";
   };
 
-  buildInputs = [ zlib tbb ];
+  buildInputs = [ zlib tbb python perl ];
```

Comment:
This won't fix the issue. Now you have `python` and `perl` available at build time, but that won't help you as they won't be there during runtime. You can test this with something like `nix-shell `pwd` --pure -p bowtie2`.

In order to properly fix the issue I recommend to patch all impure references, e.g. with `substituteInPlace`.

Reply:
It might fix shebangs though, if that's the case.

Reply:
I checked it out and tested it with the command I posted above and it broken as it couldn't find `perl`.

Reply:
Strange, it does patch the shebangs which makes it available at runtime for me?

```
nix-shell -p bowtie2 --pure --command 'bowtie2 2>&1 | head'              /tmp/result/bin
No index, query, or output file specified!
Bowtie 2 version 2.3.4.3 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)
Usage:
  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i>} [-S <sam>]

  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).
             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.
  <m1>       Files with #1 mates, paired with files in <m2>.
             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).
  <m2>       Files with #2 mates, paired with files in <m1>.

```

Reply:
Just a bit reluctant to reimplement the shebang patching, better to use the mechanisms that are already there.

Reply:
full ack. I'm afraid I used the wrong Nix path when opening the shell, sorry!